webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/TableIndexPage.js

```diff
@@ -0,0 +1,251 @@
+  const {
+    response: { search: apiSearchQuery, can_create: canCreate },
+    status = STATUS.PENDING,
+  } = useAPI('get', apiUrl, memoOptions);
+
+  const dispatch = useDispatch();
+  const memoDefaultSearchProps = useMemo(
+    () => getControllerSearchProps(controller),
+    [controller]
+  );
+  const searchProps = customSearchProps || memoDefaultSearchProps;
+
+  const setAutocomplete = useCallback(
+    newSearch => {
+      dispatch(
+        getResults({
+          url: searchProps.autocomplete.url,
+          searchQuery: newSearch,
+          controller: searchProps.controller,
+          trigger: TRIGGERS.INPUT_CHANGE,
+          id: searchProps.autocomplete.id,
+        })
+      );
+    },
+    [searchProps, dispatch]
+  );
+  useEffect(() => {
+    let newSearch;
+    if (search !== undefined) {
+      const urlParams = new URLSearchParams(search);
+      newSearch = urlParams.get('search');
+    } else {
+      newSearch = getURIsearch();
+    }
+    setAutocomplete(newSearch || '');
+    setParams({ search: newSearch });
+  }, [search, apiSearchQuery, setAutocomplete]);
+
+  const setSearch = newSearch => {
+    if (history) {
+      const uri = new URI();
+      uri.setSearch(newSearch);
+      history.push({ search: uri.search() });
+    } else {
+      changeQuery(params);
+    }
+    setParams(newSearch);
+  };
+  const onBookmarkClick = newSearch => {
+    setAutocomplete(newSearch);
+    setSearch({ search: newSearch, page: 1 });
+  };
+
+  const actionButtons = [
+    creatable &&
+      canCreate && {
+        title: __('Create new'),
+        action: { href: customCreateURL || createURL() },
+      },
+    exportable && {
+      title: __('Export'),
+      action: { href: customExportURL || exportURL() },
+    },
+    hasHelpPage && {
+      title: __('Documentation'),
+      icon: <QuestionCircleIcon />,
+      action: { href: customHelpURL || helpURL() },
+    },
+    ...customActionButtons,
+  ].filter(item => item);
+
+  return (
+    <div className="pf-c-page foreman-page">
+      <Head>
+        <title>{header}</title>
+      </Head>
+      {breadcrumbOptions && (
+        <PageSection variant={PageSectionVariants.light} type="breadcrumb">
+          <BreadcrumbBar {...breadcrumbOptions} />
+        </PageSection>
+      )}
+      <PageSection variant={PageSectionVariants.light}>
+        <TextContent>
+          <Text component="h1">{header}</Text>
+        </TextContent>
+      </PageSection>
+      {beforeToolbarComponent}
+      <PageSection variant={PageSectionVariants.light}>
+        <Toolbar>
+          <ToolbarContent>
+            {searchable && (
+              <ToolbarGroup>
+                <ToolbarItem>
+                  <SearchBar
+                    data={searchProps}
+                    initialQuery={apiSearchQuery}
+                    onSearch={newSearch =>
+                      setSearch({ search: newSearch, page: 1 })
+                    }
```

Comment:
can the consumer pass `onSearch` callback so in the parent component they could manage the search change as well and trigger an API call in the table for example.